### DISCUSSION TITLE
.vulnerabilities[].advisories[].url does not match the iri-reference pattern

### DISCUSSION THREAD

**Discussion by black-snow** [aquasecurity/trivy#6663] [Bugs]
### Description

BOM validation reports an invalid url and, in fact, trivy seems to have generated:

```json
"url": "https://github.com/jpadilla/pyjwt/commit/9c528670c455b8d948aff95ed50e22940d1ad3fc (2.4.0)"
```

(an IRI must not contain spaces)

### Desired Behavior

Should be a valid IRI

### Actual Behavior

Invalid IRI generated

### Reproduction Steps

```bash
1. set up a project containing pyjwt@1.7.1
2. generate an OCI image from it (fs should work, too - haven't checked)
3. run `trivy` with `--format=cyclonedx` on it and check the generated SBOM
```

### Target

Container Image

### Scanner

None

### Output Format

CycloneDX

### Mode

Standalone

### Debug Output

```bash
It runs within our pipeline. I can debug manually if required.
```

### Operating System

macOS, Ubuntu

### Version

```bash
0.51.1
```

### Checklist

- [ ] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Ubuntu contains the invalid URL in their advisory.
https://github.com/aquasecurity/vuln-list/blob/88661e296323aecce7cb46f8123f9072845c2c11/ubuntu/2022/CVE-2022-29217.json#L8

**Reply by black-snow** [reply]
Dang, thanks. Any idea where to report this?

I wonder if this still classifies as a bug, though. If you ask trivy to generate a cyclonedx report and it doesn't, for whatever reason, this does taste like a bug to me. What would be a sane alternative? Fail altogether? Barely. Omitting the erroneous entry with a warning, I guess.

**Reply by black-snow** [reply]
For reference: https://ubuntu.com/security/CVE-2022-29217

**Reply by knqyf263** [reply]
If you click a reference in [the Ubuntu page](https://ubuntu.com/security/CVE-2022-29217), it returns 404. It's a bug on their side. However, it is possible to add processing for such corrupted data on the Trivy side.

**Reply by black-snow** [reply]
As opposed to [ubuntu](https://ubuntu.com/security/CVE-2022-29217), [for debian](https://security-tracker.debian.org/tracker/CVE-2022-29217) it's done correctly.

**Reply by knqyf263** [reply]
Yes, mistake, ubuntu.

**Comment by knqyf263** [comment]
Created https://github.com/aquasecurity/trivy/issues/6801
